RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterQuickFix.cs

```diff
@@ -47,17 +49,20 @@ public override void Fix()
 
             SetLocalCopyVariableName();
 
-            if (IsCancelled) { return; }
+            if (!ProposedLocalVariableNameIsValid()
+                || IsCancelled)
+            {
+                return;
+            }
 
             ModifyBlockToUseLocalCopyVariable();
         }
 
         private void SetLocalCopyVariableName()
         {
-            using (var view = new AssignedByValParameterQuickFixDialog(_originalProcCodeLines))
+            using (var view = new AssignedByValParameterQuickFixDialog(_target, Selection))
```

Comment:
This hard-wired dependency is going to be a problem with unit tests: we simply cannot have a UI pop up in the middle of an automated test suite running on a CI server.